php/libraries/CouchDB.class.inc

```diff
@@ -735,5 +737,27 @@ class CouchDB
 
         return $this->_postRelativeURL('_bulk_docs', $docs);
     }
+
+    /**
+     * Validation of CouchDB Host and Port.
+     * Throws LorisException if (Host or Port) invalid.
+     *
+     * @throws LorisException
+     *
+     * @return void
+     */
+    function validHostAndPort()
+    {
+        $config       = NDB_Config::singleton();
+        $couch_config = $config->getSetting('CouchDB');
+        if (!filter_var(gethostbyname($couch_config['hostname']), FILTER_VALIDATE_IP)
+            || !is_numeric($couch_config['port'])
```

Comment:
what is port isn't set? is_numeric seems to be falsey in PHP. Maybe this should coalesce to the default CouchDB port?